File: lesson6/python3/homework/homework6_8.py
#!/usr/bin/python3
#: py3/lesson6/homework/homework6_8.py
# Напишите функцию, которая вычисляет N-ое число Фибоначчи.


def fibonacci_numb(numb):
    """Возвращает число Фибоначчи"""

    fib_1 = 0
    fib_2 = 1

    if numb == 0:
        return 0
    if numb == 1 or numb == 2:
        return 1

    for i in range(numb):
        fib_1 = fib_2 + fib_1
        fib_2 = fib_1 - fib_2

    return fib_1


fib_numb = int(input("Введите число - "))

result_fib = fibonacci_numb(fib_numb)
print(result_fib)

File: lesson6/c_c++/homework/homework6_8.c
/* 
 * py3/c_c++/lesson6/homework/homework6_8.c
 */
#include <stdio.h>

int number_fibonacci(int numb) {
    int fib_1 = 0;
    int fib_2 = 1;

    if (numb == 0) return 0;
    if (numb == 1 || numb == 2) return 1;

    int i;
    for (i = 1; i <= numb; ++i) {
        fib_1 = fib_2 + fib_1;
        fib_2 = fib_1 - fib_2;
    }

    return fib_1;
}

int main() {
    int numb_i;
    printf("Введите число - ");
    scanf("%d", &numb_i);

    int result_fib;
    result_fib = number_fibonacci(numb_i);
    printf("%d\n", result_fib);

    return 0;
}
